setup_undercloud.yml

```diff
@@ -4,39 +4,58 @@
       badfish_venv: "{{ ansible_user_dir }}/badfish/.venv"
       chassis_password:  "{{ instackenv_content.nodes[0].pm_password }}"
   tasks:
-    - name: check if os installed as root user
-      shell: |
-        sshpass -p {{ ansible_ssh_pass }}  ssh -o 'PreferredAuthentications=password' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -q root@{{ undercloud_hostname }} 'cat /etc/redhat-release'
-      register: rhel_release
-      ignore_errors: true
+    - name: os_version not set by user
+      block:
+        - name: check if os installed as root user
+          shell: |
+            sshpass -p {{ ansible_ssh_pass }}  ssh -o 'PreferredAuthentications=password' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -q root@{{ undercloud_hostname }} 'cat /etc/redhat-release'
+          register: rhel_release
+          ignore_errors: true
 
-    - name: set rhel version
-      set_fact:
-        rhel_version: "{{ rhel_release.stdout.split()[-2] }}"
-      ignore_errors: true
-      when: rhel_release.stdout != ""
+        - name: set rhel version
+          set_fact:
+            rhel_version: "{{ rhel_release.stdout.split()[-2] }}"
+          ignore_errors: true
+          when: rhel_release.stdout != ""
 
-    - name: Check if os is installed as stack user
-      shell: |
-        ssh -o 'PreferredAuthentications=publickey' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -i ~/.ssh/id_rsa stack@{{ undercloud_hostname }} 'cat /etc/redhat-release'
-      register: rhel_release
-      ignore_errors: true
-      when: rhel_version is not defined
+        - name: Check if os is installed as stack user
+          shell: |
+            ssh -o 'PreferredAuthentications=publickey' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -i ~/.ssh/id_rsa stack@{{ undercloud_hostname }} 'cat /etc/redhat-release'
+          register: rhel_release
+          ignore_errors: true
+          when: rhel_version is not defined
 
-    - name: set rhel version
-      set_fact:
-        rhel_version: "{{ rhel_release.stdout.split()[-2] }}"
-      when: rhel_version is not defined
+        - name: set rhel version
+          set_fact:
+            rhel_version: "{{ rhel_release.stdout.split()[-2] }}"
+          when: rhel_version is not defined
 
-    - name: set os_install
-      set_fact:
-        os_install: "{{ (((osp_release|int > 14) and (rhel_version is version('8.0', '<')))|ternary('RHEL 8.1', ((osp_release|int < 15) and (rhel_version is version ('8.0', '>=')))|ternary('RHEL 7.7', false))) }}"
-      when: force_reprovision == false
+        - name: set os_install
```

Comment:
NIT: all three force_reprovision = false tasks can be moved under the single block (adds readability)